instrumentation/jetty-httpclient/jetty-httpclient-9.0/library/src/test/groovy/io/opentelemetry/javaagent/instrumentation/jetty/httpclient/v9_0/JettyHttpClient9LibraryTest.groovy

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jetty.httpclient.v9_0
+
+import io.opentelemetry.context.Context
+import io.opentelemetry.instrumentation.test.LibraryTestTrait
+import org.eclipse.jetty.client.api.Request
+
+import static io.opentelemetry.javaagent.instrumentation.jetty.httpclient.v9_0.JettyHttpClient9Tracer.tracer
+
+class JettyHttpClient9LibraryTest extends AbstractJettyClient9Test implements LibraryTestTrait {
+
+
+  void attachInterceptor(Request jettyRequest, Context parentContext) {
+
+    if (!tracer().shouldStartSpan(parentContext)) {
+      return;
+    }
+
+    JettyHttpClient9TracingInterceptor interceptor = new JettyHttpClient9TracingInterceptor(parentContext);
```

Comment:
Have you tried adding interceptor to whole httpclient? From https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/client/HttpClient.html#getRequestListeners() seem it should be possible. Perhaps interceptor then could start span in `onBegin​` you could attach `Context` as request attribute so you could have access to it when request finishes. If this works then try the same approach with javaagent instrumentation, instead of instrumenting `send` attach your interceptor to http client when it is created.

Reply:
Thanks @laurit.  very good observation.   I don't have a interceptor on the whole HttpClient.  I have a working version of this in use today that intercepts at the client.getRequest() which would be one step below what you are suggesting.   From looking at some of the Jetty9 source code just now, I see the request listeners you are showing here are on the HttpClient  are applied on any request.  There is another set of listeners that are held at the actual request object instance.  I do see the appeal of applying at that top level and it may mean that we could avoid missing span since this *should* catch internal spans that emanate from either the listeners on the HttpClient or the listeners on the Request object.   I am in process of changing my code to wrap all important listeners as suggessted by @anuraaga.   So, let me leave open the conversation and I will try this suggestion

Reply:
I ended up using HttpClient, but found there are two layers of Request Listeners, using the second set of listeners which are part of the Request only